shaders/src/main.vs

```diff
@@ -35,9 +35,21 @@ void main() {
         // because we ensure the worldFromModelNormalMatrix pre-scales the normal such that
         // all its components are < 1.0. This prevents the bitangent to exceed the range of fp16
         // in the fragment shader, where we renormalize after interpolation
-        vertex_worldTangent.xyz = objectUniforms.worldFromModelNormalMatrix * vertex_worldTangent.xyz;
         vertex_worldTangent.w = mesh_tangents.w;
         material.worldNormal = objectUniforms.worldFromModelNormalMatrix * material.worldNormal;
+
+        // Compute the tangent according to Frisvad's paper
+        // see: https://backend.orbit.dtu.dk/ws/portalfiles/portal/126824972/onb_frisvad_jgt2012_v2.pdf
+        // We need to swap axes compared to the paper, to have the singularity at -Y (bottom in Filament coordinates)
+        // Handle the singularity
+        if (material.worldNormal.y < -0.9999999) {
+            vertex_worldTangent.xyz = vec3(0.0, 0.0, -1.0);
+        } else {
+            float a = 1.0 / (1.0 + material.worldNormal.y);
+            float b = -material.worldNormal.x * material.worldNormal.z * a;
+            vertex_worldTangent.xyz = vec3(1.0 - material.worldNormal.x * material.worldNormal.x * a, b, -material.worldNormal.x);
+        }
```

Comment:
This incurs extra vertex shader operations in the app, not just in GLTF viewer, right? Can you update the tangent calculation on the mesh import side instead?

Reply:
Actually, I've just realized that this would mean that we are diverging the GltfViewer and Shapr3D codebase. Please use macros instead (in particular, `IN_SHAPR_SHADER` for this case) so that we never ever again have to maintain two parallel code bases.

Please also PR-up a fixup in a similar vein to the normal map fix, that is, use `defined(IN_SHAPR_SHADER)` and `!defined(IN_SHAPR_SHADER)` to differentiate between XYZ and XZY. Only our shader compilation is defining `IN_SHAPR_SHADER`, we are not setting it for the viewer ever.

Reply:
I fixed the Filament side here: [c18f501](https://github.com/shapr3d/filament/pull/36/commits/c18f50192286250160cc8085a904c957984a9b37). This will make Filament compute the tangents in the vertex shader, while Shapr3D will use the ones computed using the same method, but at tessellation time.

Reply:
Please don't forget to fix up the triplanar in GltfViewer and create a ticket for that. As for this particular PR, things look good to me.